src/cluster.jl

```diff
@@ -161,15 +163,16 @@ function check_worker_state(w::Worker)
         else
             w.ct_time = time()
             if myid() > w.id
-                t = @async exec_conn_func(w)
+                t = Threads.@spawn exec_conn_func(w)
```

Comment:
I think given the tasks created by Distributed are generally orchestration and not work, I think they should stay on the same threadpool as the caller, which is likely thread 1, and may be `:interactive` so I think these all should be
```suggestion
                t = Threads.@spawn Threads.threadpool() exec_conn_func(w)
```
for which I've proposed shorthand of `@spawn :same` here https://github.com/JuliaLang/julia/pull/57109

Reply:
True, but then why ensure that they stay on the same threadpool as the caller if they're generally lightweight orchestration tasks?

Reply:
I guess because the worker threads could be tied up doing blocking stuff.

Reply:
Ok, I set most tasks to use `Threads.threadpool()` in 7f1d879. Some I didn't because they're intended to be executed on a worker, which I feel should 'own' the worker process and not have to be sensitive to any non-Distributed code.